src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts

```diff
@@ -145,32 +145,50 @@ export class ContentTypeFieldsPropertiesFormComponent extends BaseComponent impl
 
     }
 
-    private handleAutoCheckValues(): void {
+    private setAutoCheckValues(): void {
         if (this.form.get('searchable')) {
-            this.form.get('searchable').valueChanges.subscribe(res => this.setIndexedValueChecked(res));
+            this.handleCheckValues(this.form.get('searchable'));
         }
         if (this.form.get('listed')) {
-            this.form.get('listed').valueChanges.subscribe(res => this.setIndexedValueChecked(res));
+            this.handleCheckValues(this.form.get('listed'));
         }
         if (this.form.get('unique')) {
-            this.form.get('unique').valueChanges.subscribe(res => this.handleUniqueValuesChecked(res));
+            this.handleCheckValues(this.form.get('unique'));
         }
     }
 
+    private handleCheckValues(checkbox: any): void {
+        if (checkbox.value) {
+            if (checkbox === this.form.get('unique')) {
+                this.handleDisabledRequired(true);
+            }
+            this.handleDisabledIndexed(true);
+        }
+
+        checkbox === this.form.get('unique') ? checkbox.valueChanges.subscribe(
```

Comment:
Two subscribe are unnecessary here, I think you can do:

```
checkbox.valueChanges.subscribe(res => {
    checkbox === this.form.get('unique') ? this.handleUniqueValuesChecked(res) : this.setIndexedValueChecked(res)
});
``` 

And it's more readable too.